test/perf/perf_alternator_workloads.cc

```diff
@@ -0,0 +1,286 @@
+                },
+                ":C4": {
+                    "M": {"Name": {"S": "Joe"}, "Age": {"N": "35"}}
+                },
+                ":C5": {
+                    "N": "123.45"
+                },
+                ":C6": {
+                    "NS": ["42.2", "-19", "7.5", "3.14"]
+                },
+                ":C7": {
+                    "NULL": true
+                },
+                ":C8": {
+                    "S": "Hello"
+                },
+                ":C9": {
+                    "SS": ["Giraffe", "Hippo" ,"Zebra"]
+                }
+            },
+            "ReturnValues": "NONE"
+        }
+    )";
+    return make_request(cli, "UpdateItem", prefix+suffix);
+}
+
+static future<> get_item(http::experimental::client& cli, uint64_t seq) {
+    auto body = format(R"({{
+        "TableName": "workloads_test",
+        "Key": {{
+            "p": {{
+                "S": "{}"
+            }},
+            "c": {{
+                "S": "{}"
+            }}
+        }},
+        "ProjectionExpression": "C0, C1, C2, C3, C4, C5, C6, C7, C8, C9",
+        "ConsistentRead": false,
+        "ReturnConsumedCapacity": "TOTAL"
+    }})",seq, seq);
+    co_await make_request(cli, "GetItem", std::move(body));
+}
+
+static void flush_table() {
+    auto cli = get_client(10000);
+    auto req = http::request::make("POST", "localhost", "/storage_service/keyspace_flush/alternator_workloads_test");
+    cli.make_request(std::move(req), [] (const http::reply& rep, input_stream<char>&& in) {
+        return make_ready_future<>();
+    }).get();
+    cli.close().get();
+}
+
+static void create_partitions(const test_config& c, http::experimental::client& cli) {
+    std::cout << "Creating " << c.partitions << " partitions..." << std::endl;
+    for (unsigned seq = 0; seq < c.partitions; ++seq) {
+        update_item(cli, seq).get();
+    }
+    if (c.flush) {
+        std::cout << "Flushing partitions..." << std::endl;
+        flush_table();
+    }
+}
+
+void workload_main(const test_config& c) {
+    std::cout << "Running test with config: " << c << std::endl;
+
+    auto cli = get_client(c.port);
+    auto finally = defer([&] {
+        delete_alternator_table(cli);
+        cli.close().get();
+    });
+
+    create_alternator_table(cli);
+    using fun_t = std::function<future<>(http::experimental::client&, uint64_t)>;
+    std::map<std::string, fun_t> workloads = {
+        {"read",  get_item},
+        {"write", update_item},
+    };
+
+    if (c.workload == "read") {
+        create_partitions(c, cli);
+    }
+
+    auto it = workloads.find(c.workload);
+    if (it == workloads.end()) {
+        throw std::runtime_error(format("unknown workload '{}'", c.workload));
+    }
+    fun_t fun = it->second;
+
+    auto results = time_parallel([&] {
+        static thread_local auto sharded_cli = get_client(c.port); // for simplicity never closed as it lives for the whole process runtime
```

Comment:
I don't understand what I'm seeing here. You have "concurrency", and yet just one "cli" object per shard. How does it work?

Reply:
I think it was working because underneath `connection::make_request` doesn't yield, at least in this case. 

Although I think it's more realistic for alternator  to work on a higher number of connections, so I will add a pool, making one number of connections equal to concurrency.